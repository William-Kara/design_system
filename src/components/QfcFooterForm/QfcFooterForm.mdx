import { ArgTypes, Meta } from '@storybook/blocks';
import QfcFooterForm from '@/components/QfcFooterForm/QfcFooterForm.vue';

<Meta title="Modules/FooterForm" />

# Definition

QfcFooterForm is used in form pages that requiers minimal footer information.

## Props

The following props are available for the QfcFooter component:

<ArgTypes of={QfcFooterForm} />

# Usage

To use QfcFooter, style needs to be imported separately.

```sass
@import '@website/design-system-vuejs/dist/styles/bulma/layout/footer.scss';
@import '@website/design-system-vuejs/dist/styles/components/QfcFooterForm.scss';
```

```html
<template>
  <QfcFooterForm >
</template>
```
