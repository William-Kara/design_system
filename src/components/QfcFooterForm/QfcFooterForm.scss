.footer {
  padding: 0 18rem;
  font-size: 1.12rem;
  .footer__link {
    display: block;
  }
}

@include mobile {
  .footer {
    padding-top: 2rem;
    margin-left: 2rem;
    padding-bottom: 5rem;
  }
  .footer {
    margin-left: 0;
    margin-right: 2rem;
  }
}
