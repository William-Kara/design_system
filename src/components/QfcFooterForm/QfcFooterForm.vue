<script lang="ts" setup>
import QfcLink from './../QfcLink';

interface Links {
  links: Link[];
}
interface Link {
  to: string;
  isExternalLink: boolean;
  iconSrc: string;
}
interface Props {
  firstLinks: Links;
  secondLinks: Links;
  thirdLinks: Links;
}
withDefaults(defineProps<Props>(), {
  firstLinks: () => ({
    links: [
      {
        label: 'Mentions légales',
        to: '/mentions-legales',
        isExternalLink: false,
      },
      {
        label: 'Politique de confidentialité',
        to: '/politique-de-confidentialite',
        isExternalLink: false,
      },
      {
        label: 'Gestion des cookies',
        to: '/gestion-des-cookies',
        isExternalLink: false,
      },
    ],
  }),
});
</script>
<template>
  <footer class="footer py-6">
    <div class="columns container has-text-centered">
      <div class="column footer__column">
        <template v-for="({ label, to, isExternalLink }, index) in firstLinks.links" :key="index">
          <QfcLink :label="label" :href="to" :is-external-link="isExternalLink" :iconSrc="iconSrc">
          </QfcLink>
        </template>
      </div>
      <div class="column footer__column">
        <template v-for="({ label, to, isExternalLink }, index) in secondLinks.links" :key="index">
          <QfcLink :label="label" :href="to" :is-external-link="isExternalLink" :iconSrc="iconSrc">
          </QfcLink>
        </template>
      </div>
      <div class="column footer__column">
        <template v-for="({ label, to, isExternalLink }, index) in thirdLinks.links" :key="index">
          <QfcLink :label="label" :href="to" :is-external-link="isExternalLink" :iconSrc="iconSrc">
          </QfcLink>
        </template>
      </div>
    </div>
  </footer>
</template>

<style lang="scss" scoped src="./QfcFooterForm.scss" />
