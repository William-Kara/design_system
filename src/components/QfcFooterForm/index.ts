import QfcFooter from './QfcFooterForm.vue';

export { QfcFooter };
export default QfcFooter;
