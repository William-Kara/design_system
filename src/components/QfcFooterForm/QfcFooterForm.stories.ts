import QfcFooterForm from './QfcFooterForm.vue';
import type { Meta, StoryObj } from '@storybook/vue3';

const meta: Meta<typeof QfcFooterForm> = {
  component: QfcFooterForm,
  title: 'Modules/FooterForm',
};

export default meta;
type Story = StoryObj<typeof meta>;

export const Primary: Story = {
  args: {
    firstLinks: {
      links: [
        {
          label: 'Mentions légales',
          to: '/',
          isExternalLink: false,
          isButton: false,
        },
      ],
    },
    secondLinks: {
      links: [
        {
          label: 'Politique de confidentialité',
          to: '/',
          isExternalLink: false,
          isButton: false,
        },
      ],
    },
    thirdLinks: {
      links: [
        {
          label: 'Gestion des cookies',
          to: '/',
          isExternalLink: false,
          isButton: false,
        },
      ],
    },
  },
};
