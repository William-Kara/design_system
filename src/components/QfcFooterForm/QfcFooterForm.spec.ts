import { it, describe, beforeEach, expect } from 'vitest';

import { shallowMount } from '@vue/test-utils';
import QfcFooterForm from './QfcFooterForm.vue';

let wrapper: any = null;

const propsValues = {
  firstLinks: {
    links: [
      {
        label: 'link 1',
        to: '/link-1',
        isExternalLink: false,
      },
    ],
  },
  secondLinks: {
    links: [
      {
        label: 'link 2',
        to: '/link-2',
        isExternalLink: false,
      },
    ],
  },
  thirdLinks: {
    links: [
      {
        label: 'link 3',
        to: '/link-3',
        isExternalLink: false,
      },
    ],
  },
};

beforeEach(() => {
  wrapper = shallowMount(QfcFooterForm, {
    propsData: propsValues,
  });
});

describe('QfcFooterForm Component', () => {
  it('should mount properly', () => {
    expect(wrapper).toBeTruthy();
  });

  it('should have correct props', () => {
    expect(wrapper.props()).toEqual(propsValues);
  });

  it('should render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
