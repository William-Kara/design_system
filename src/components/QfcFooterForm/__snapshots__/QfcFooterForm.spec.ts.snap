// Vitest Snapshot v1

exports[`QfcFooterForm Component > should render correctly 1`] = `
"<footer data-v-843c6c56=\\"\\" class=\\"footer py-6\\">
  <div data-v-843c6c56=\\"\\" class=\\"columns container has-text-centered\\">
    <div data-v-843c6c56=\\"\\" class=\\"column footer__column\\">
      <qfc-link-stub data-v-843c6c56=\\"\\" label=\\"link 1\\" href=\\"/link-1\\" isexternallink=\\"false\\" underline=\\"false\\"></qfc-link-stub>
    </div>
    <div data-v-843c6c56=\\"\\" class=\\"column footer__column\\">
      <qfc-link-stub data-v-843c6c56=\\"\\" label=\\"link 2\\" href=\\"/link-2\\" isexternallink=\\"false\\" underline=\\"false\\"></qfc-link-stub>
    </div>
    <div data-v-843c6c56=\\"\\" class=\\"column footer__column\\">
      <qfc-link-stub data-v-843c6c56=\\"\\" label=\\"link 3\\" href=\\"/link-3\\" isexternallink=\\"false\\" underline=\\"false\\"></qfc-link-stub>
    </div>
  </div>
</footer>"
`;
